### DISCUSSION TITLE
Welcome to rikaikun Discussions!

### DISCUSSION THREAD

**Discussion by melink14** [melink14/rikaikun#363] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by 1tomadeira** [comment]
Hello, hope you're doing fine.

I had the extension working completly fine, yet for some reason it ended all being all white (happened me once already one year ago, no more), and tried to reinstall the extension again. Unfortunely I noticed it was removed from Opera Store. Yes there is an option to use Chrome Extensions, but many use the Opera to avoid using Google stuff. 

I understand your decision, but will the Opera version be discontinued?

Best Regards.

**Reply by melink14** [reply]
Hi!

Sorry to hear about the extension being removed from the Opera store. I actually had nothing to do with it being uploaded there and I'm not sure who did it or why they chose to remove it later.

That being said, it seems like it's pretty common to install extensions meant for Google Chrome on Opera. I sympathize with wanting to avoid Google stuff but in this case you can consider chrome extensions not Google stuff but community stuff that happens to work on Google software!

**Comment by SamDunlap** [comment]
Hello! I really enjoy using this extension to assist with some of my school work if it contains kanji I have not learned.

For a school project I am required to create and submit a bug fix for an open source project, and I would love to help out this project, as it has been so useful to me over my college career.

I see that some of the issues are tagged with "good first issue" already and I am curious if this is a good place to start.

Thanks!

**Reply by melink14** [reply]
Hi!

I'm glad it's been useful for you and I think you can definitely help out.

As you you noticed, the "good first issue" tag is being used for issues which are relatively small and don't require full context to start work. They range a bit in size though so be careful if your school project has some kind of requirement for complexity or 'lines of code'.

It got bogged down in editing and code review, but feel free to check out the readme and contributing guide I drafted last fall in this PR:
https://github.com/melink14/rikaikun/pull/261

Unfortunately, I can't figure out how to view markdown files normally while in PR but they're not too hard to read even in markdown mode (Here's direct link to CONTRIBUTING.md):
https://github.com/melink14/rikaikun/pull/261/files?file-filters%5B%5D=.md#diff-eca12c0a30e25b4b46522ebf89465a03ba72a03f540796c979137931d8f92055

**Reply by SamDunlap** [reply]
I have no such requirements, only that there is some process of getting changes accepted that I need to write about.
Regardless, I can always try to work on more issues if I feel I haven't contributed sufficiently.

As a side note, I think you click on the kebab menu for the file on the pull request and then click view file to show the formatted version of the file.

Thanks so much for the info and the quick response.

**Reply by melink14** [reply]
> As a side note, I think you click on the kebab menu for the file on the pull request and then click view file to show the formatted version of the file.

This appeared to try to link to the file on the master branch which doesn't exist due to the file being new. I did see that if you use 'rich diff' mode it does render mostly with some extra coloring to show 'new' parts.

Anyway, as I mentioned before, for how to contribute please read that draft contribution guide in that PR. Here's the direct link to the formatted section:
https://github.com/melink14/rikaikun/diffs/1?base_sha=d1d283077c0d22f0f358d6d0facb751cab0dfd44&commentable=true&pull_number=261&sha1=d1d283077c0d22f0f358d6d0facb751cab0dfd44&sha2=6b13efb78b6985191708ff70307eb7a05bb1026e&short_path=eca12c0&unchanged=expanded#code-contributions

(Though that link hides the style guides so here's the full guide: [Contributing Code](https://github.com/melink14/rikaikun/pull/261/files?file-filters%5B%5D=.md&short_path=eca12c0#diff-eca12c0a30e25b4b46522ebf89465a03ba72a03f540796c979137931d8f92055))

**Reply by SamDunlap** [reply]
I'm not exactly sure where to ask this, but I have gotten my environment set up so I can build with no issues and can add the unpacked extension to chrome, but when I click on it, I am not greeted with the "on" icon, is this a known thing with the master branch/do you have a suggestion?

**Reply by SamDunlap** [reply]
TLDR; I fixed it by removing all multiline comments, I will certainly revert this change for any PR's made, but multiline comments were seen as an unknown token by Chrome for some reason.

**Reply by melink14** [reply]
(You can keep discussing here or make your own topic(s) in the dicussion group if you want)

I hadn't seen this; I wonder if it's something that's only happening with newer versions of chrome (since it's been awhile since I've tested loading unpacked after building...)

Can you open a bug with a pointer with the full details you found? No worries if not, I'll also try to remember to check later!

**Reply by SamDunlap** [reply]
Yeah I can take care of that at some point today. I had the same issue with both Chrome and chromium based Edge. If I run into anything else I'll just start a new discussion topic.

**Reply by SamDunlap** [reply]
I have fixed the issue linked here https://github.com/melink14/rikaikun/issues/346. I don't seem to have the ability to create a branch related to the issue. How should I proceed?

**Reply by melink14** [reply]
(FYI, I filed the issue about synatx errors with comments here: https://github.com/melink14/rikaikun/issues/455; it seems to be an encoding problem so not sure why removing comments fixed it. :D)

**Comment by melink14** [comment]
Thanks!

This might be a failing in the docs but maybe you didn't create a fork
first? That's generally the first step, then you can open a pull request
with your change.

I'm thinking my contribution docs may need a few edits!

2021年4月22日(木) 11:51 Samuel Dunlap ***@***.***>:

**Reply by SamDunlap** [reply]
Yeah I don't seem to have the ability to do that either. I created a branch locally, but I cannot push that, nor can I create one on the web interface.

**Reply by melink14** [reply]
Looking at your account it seems like you haven't forked it yet so I should be a bit more explicit.

By fork, I mean the github notion of forking which creates a copy of the repository in your  account. (For example, a past contributor: https://github.com/darren-lester/rikaikun)

There's a button to fork in the top right of the main repo page but you can see a guide here:
https://app.egghead.io/lessons/javascript-how-to-fork-and-clone-a-github-repository?pl=how-to-contribute-to-an-open-source-project-on-github

(The other sections might be useful as well)

You can't make a branch directly due to permission issues but you can treat a fork as one huge branch owned by you. :)

**Reply by SamDunlap** [reply]
Thanks for all the help! I think I finally got it squared away. I haven't used GitHub as much as I probably should have in my college career, but I suppose this is good practice. I'm assuming you will let me know if you would like any of my changes edited in the PR itself.

**Reply by melink14** [reply]
It's no problem; it's never to early to start! See you over in the PR!

**Comment by SamDunlap** [comment]
I was looking to fix this bug too, I have run into this bug myself, but I can't seem to find the detection logic. Can you point me in the right direction?

Issue: https://github.com/melink14/rikaikun/issues/215

--I actually just copied the div from the issue into an html file and rikaikun detected it.
--I also can't replicate on Twitch.tv, so it must be resolved?

**Reply by SamDunlap** [reply]
I'm still curious about where the detection logic is though, so I can fix the issue with google docs as well.

Issue: https://github.com/melink14/rikaikun/issues/213

**Reply by melink14** [reply]
Detection code starts here: https://github.com/melink14/rikaikun/blob/a2cd8c16cbc705a80238758f5a984b4e0046d239/extension/rikaicontent.ts#L914

Basically, whenever the mouse moves this function is called and it tries to see if there's text to look up at that point.

I will be honest, the detection code is some of the hardest code to work with since I've not refactored it in many years and it has many edge cases to handle bugs. Plus since there are no tests for rikaikun, you have to constantly worry that you broke a rare edge case when making changes.

Ideally, I hope to re-write it with modern practices so it's easier to maintain and edit but just know that now it might end up being harder than you think and it might require more discussion in the PR.

I'd start with putting break points in various places and seeing the code flow in the broken case. Also, we should try to figure out what makes the broken HTML not work in general and see if we can fix it in the most general way possible; we don't want a solution that *only* works for google docs if we can help it.

**Reply by SamDunlap** [reply]
For sure, I will definitely take a look. I'll submit a PR early so I am on track.

**Comment by Zkarts** [comment]
Hey, I got the built extension to run with the info above, which was great, but I wasn't able to figure out where to look for my issue. Heck, I don't even know if it would be possible.
Basically, I have a Japanese epub that I want to try reading through an epub reader extension with the help of rikaikun. That leads to a url in the form of `chrome-extension://jhhclmfgfllimlhabjkgkeebkbiadflb/reader.html?filename=file`. I managed to figure out rikaikun apparently doesn't have access to the page as it says "cannot read or write site data" or something like that when I right-click the rikaikun icon. I'm not sure if it would be possible to grant rikaikun access to that page/URL, but I also don't really know where to look as I've never worked on chrome extensions before.

**Reply by melink14** [reply]
Hi!

This has come up before and I filed an issue here: https://github.com/melink14/rikaikun/issues/381 which you can follow and collaborate on. IIRC, there were some old posts saying it was possible but nothing verified recently!

In the meantime, also consider this web only e-book reader some bloggers have liked:
https://ttu-ebook.web.app/

**Reply by Zkarts** [reply]
Thanks for chasing this down. I thought I saw some similar issues that said this was fixed, so I got confused when I didn't get it to work.
Maybe it does make sense for them not to allow it based on extensions stealing other extensions' permissions that way, but cool to see you filed a report for it :)

Until then the web-based e-book reader sounds like a great solution to hold me over!
Only now that I try it, the text is oriented vertically and that flips rikaikun's text sideways as well 😅 
![image](https://user-images.githubusercontent.com/9387817/126072087-be4d49f5-56b8-43b3-beca-3a99bb315103.png)

**Reply by melink14** [reply]
Haha, that's amazing. I had bookmarked the site but hadn't tested it yet. 😂 

That being said something like this would be prevented by #221 but I think a faster fix would be to set `writing-mode: horizontal-tb` on the #rikaichan-window styles. (You could do that yourself via devtools in the super short term!)

Filed #613 for the eaiser fix!

**Reply by Zkarts** [reply]
I can confirm it works! The only thing I ran into was that the popup would cover most of the text that it was translating, so locally I added a margin-left of 2em to offset it a bit. It doesn't look too awful for horizontal text per se either, but it's not directly underneath the cursor anymore then. The best option I think would be to recognise the vertical writing of the base text and adding the offset depending on that.

**Reply by melink14** [reply]
Yeah, good point on the positioning. The pop up position code is super ugly (handling edge cases around if the mouse is near a screen edge.) so I might wait on that one given how easy it is to change for those in the know.

**Comment by maawisul** [comment]
Hello.

I want to contribute to this repository as a part of my school project. I am introduced to this extension just recently and have not used it that much yet but felt that this extension will help me greatly in the future with my Japanese language study.

I have taken a look at issue [898](https://github.com/melink14/rikaikun/issues/898) and [89](https://github.com/melink14/rikaikun/issues/89) and have several questions about them.

For issue [89](https://github.com/melink14/rikaikun/issues/89), can the fix be done on the dictionary or it must strictly be done on the code as I felt like the underlying issue was a missing entry from the dictionary.

Secondly, on issue [898](https://github.com/melink14/rikaikun/issues/898), the extension installed from chrome webstore worked fine for me on a document in Google Docs, however when I load the unpacked version into chrome, it didn't seem to work anymore.

Should I reply on each respective issues to figure out my next step? I am eager to help fix these issues as soon as possible.
Thanks!

**Reply by melink14** [reply]
HI @maawisul!

Thanks for your interest! Those are good questions but to your last point, I think it would be best if you added them to the issues directly so that discussion for each issue all happens in one place!

If you want, you can just copy your question to each issue and I'll answer them there!

**Reply by maawisul** [reply]
Noted.  Looking forward to the discussions

**Comment by whitesiroi** [comment]
Love the extension.
Is there a way to make it work in Notion?
thank you very much

**Reply by melink14** [reply]
I'm not familiar with Notion too much. That's the note taking app right? Does it have an extensions platform based on chrome or something similar?

**Reply by whitesiroi** [reply]
Hi @melink14, thanks for the reply!

Yeah, Notion is a note-taking and productivity app. I’m using the Notion website in Chrome (not a desktop app or anything), and I just started using it yesterday for a new job. I noticed that Rikaikun doesn’t pick up text in a few specific areas, which is why I reached out. But overall, it works perfectly fine for most of Notion — so no worries at all!

Thanks again for maintaining this awesome extension 🙏
